### DISCUSSION TITLE
[Feature] Replace Redis with Valkey

### DISCUSSION THREAD

**Discussion by ngosang** [immich-app/immich#12753] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

The idea is replace Redis with Valkey where it's possible. Eg: docker-compose, documentation...

Valkey has several advantages compared to Redis:
* It's Open Source (Redis is not anymore)
* It's maintained by a big community included big companies
* It's under active development. They are increasing the performance a lot => https://www.phoronix.com/news/Valkey-8.0-Released
* It's compatible with Redis. They provide a Docker image that is a drop-in replacement,

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
We're planning to move the job queues into postgres, at which point we'll drop this dependency entirely, so we're not planning to replace it before then.

**Reply by Nairou** [reply]
> We're planning to move the job queues into postgres, at which point we'll drop this dependency entirely, so we're not planning to replace it before then.

Came here to suggest this as well. Even SQLite has shown to be more performant than Redis in some cases (https://wafris.org/blog/rearchitecting-for-sqlite). For single-server installations, Redis can be more of a liability than a benefit.

**Comment by megvadulthangya** [comment]
@ngosang  I just replaced the redis image session  nonsesne long shakey thing with  valkey/valkey:8.0.1-alpine3.20 and works great... 5 min up till... 
if something wrong il get back here and tell it...

@bo0tzz postgres needs sse4x? Couse thats means my instance is gone :( im on a low power oldass AMD E-350

**Reply by bo0tzz** [reply]
I'm not sure what the context of your question is? Immich already requires postgres.
